File: src/main/java/com/qtech/bubbles/data/GameData.java
package com.qtech.bubbles.data;

import org.bson.BsonDocument;
import org.bson.RawBsonDocument;
import org.bson.codecs.BsonDocumentCodec;

import java.io.*;

public class GameData {
    protected RawBsonDocument document = new RawBsonDocument(new BsonDocument(), new BsonDocumentCodec());

    protected void dump(OutputStream stream) throws IOException {
        stream.write(document.getByteBuffer().array());
    }

    protected void dump(File file) throws IOException {
        FileOutputStream fos = new FileOutputStream(file);
        dump(fos);
    }

    protected void load(InputStream stream) throws IOException {
        RawBsonDocument nbt = new RawBsonDocument(stream.readAllBytes());

        this.document.clear();

        for (String key : nbt.keySet()) {
            this.document.put(key, nbt.get(key));
        }
    }

    protected void load(File file) throws IOException {
        load(new FileInputStream(file));
    }
}

File: src/main/kotlin/qtech/bubbles/data/GameData.kt
package qtech.bubbles.data

import org.bson.BsonDocument
import org.bson.RawBsonDocument
import org.bson.codecs.BsonDocumentCodec
import java.io.*

open class GameData {
    protected var document = RawBsonDocument(BsonDocument(), BsonDocumentCodec())
    @Throws(IOException::class)
    protected fun dump(stream: OutputStream) {
        stream.write(document.byteBuffer.array())
    }

    @Throws(IOException::class)
    protected fun dump(file: File) {
        val fos = FileOutputStream(file)
        dump(fos)
    }

    @Throws(IOException::class)
    protected fun load(stream: InputStream) {
        val nbt = RawBsonDocument(stream.readAllBytes())
        document.clear()
        for (key in nbt.keys) {
            document[key] = nbt[key]
        }
    }

    @Throws(IOException::class)
    protected fun load(file: File) {
        load(FileInputStream(file))
    }
}